### PR TITLE
Added a thermal hydraulics applications page

### LINKED ISSUE #26857: Make a page summarizing the various MOOSE-based TH applications
## Reason
Potential users of MOOSE-based thermal hydraulic (TH) applications have difficulty understanding the capabilities of the various applications available in this area, and may abandon the idea of using these codes as a result.

## Design
This would add a common TH page on the MOOSE website that gives descriptions of each of the TH codes and links to pages for their websites and/or validation status.

## Impact
Modified links on MOOSE website, with a new page.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #26857

### CODE REVIEW

**Comment by joshuahansel** [comment]
@lindsayad @GiudGiud @licharlot @tanoret @grmnptr @kyriv1980

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Thermal Hydraulics Applications
+
+There are several MOOSE-based applications modeling thermal hydraulics, which
+are developed as part of the
+[Nuclear Energy Advanced Modeling and Simulation (NEAMS) program](https://neams.inl.gov/).
+These applications provide capabilities at various levels of fidelity vs.
+computational efficiency for a multi-layered simulation strategy. The following
+table summarizes these applications, in order of decreasing fidelity.
+
+| Application | Description | Link(s) |
```
Comment (GiudGiud) [inline modules/doc/content/applications/thermal_hydraulics.md:10]: I would either split this table in open & not open source

also NS module should be featured?

OR add a column

**Comment by aprilnovak** [comment]
Would you be open to including Cardinal on here? For the NekRS interface?

**Comment by tanoret** [comment]
> Would you be open to including Cardinal on here? For the NekRS interface?

@aprilnovak Yes, absolutely! Could you please take a look at the current status and let us know how you would like this to be included? Thanks!

**Comment by tanoret** [comment]
I think we still need a few more examples in the applications gallery (last section). Currently it is too NS-module and reactor focused imo. If you have more diverse applications available, could you please add them? Thank you!! 

image

**Comment by GiudGiud** [comment]
Figure 5 is not NS. It s heat conduction + some THM/RELAP7 solves.

**Comment by tanoret** [comment]
> Figure 5 is not NS. It s heat conduction + some THM/RELAP7 solves. The flow over circle pic would look good in the gallery

Thanks. Would you please correct/add?

**Comment by tanoret** [comment]
Also, I am still not super happy with the site. I think that the message is not sort of clear, but not fully evident at one glance. If someone have an idea on how to make the massage pass through faster, please go ahead

**Comment by aprilnovak** [comment]
@joshuahansel here is some suggested additions for Cardinal - I don't have permissions to push to your fork, but can do that directly if you prefer.

```
## Open-Source Applications for Advanced Engineering Modeling

Open-source applications for thermal-fluid modeling provide additional capabilities.

| Application Name                                               | Based On                  | Added Features                                                                                                                                         |
| :------------------------------------------------------------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
| [Cardinal](https://cardinal.cels.anl.gov/) | [NekRS](https://github.com/Nek5000/nekRS) CFD      | CPU and GPU capabilities for RANS, LES, and DNS. Additional features include Lagrangian particle transport, an ALE mesh solver, overset meshes, and more.                                       |
```

**Comment by joshuahansel** [comment]
@aprilnovak I just sent a collab invite, so please push directly - thanks!

**Comment by aprilnovak** [comment]
Thanks @joshuahansel - added!

I can also contribute some CFD images for the gallery, if you are interested? They are mostly all reactor-focused.

**Comment by tanoret** [comment]
> @joshuahansel here is some suggested additions for Cardinal - I don't have permissions to push to your fork, but can do that directly if you prefer.
> 
> ```
> ## Open-Source Applications for Advanced Engineering Modeling
> 
> Open-source applications for thermal-fluid modeling provide additional capabilities.
> 
> | Application Name                                               | Based On                  | Added Features                                                                                                                                         |
> | :------------------------------------------------------------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
> | [Cardinal](https://cardinal.cels.anl.gov/) | [NekRS](https://github.com/Nek5000/nekRS) CFD      | CPU and GPU capabilities for RANS, LES, and DNS. Additional features include Lagrangian particle transport, an ALE mesh solver, overset meshes, and more.                                       |
> ```

@aprilnovak I am a bit worried about duplication of capabilities for RANS since we are stating that both Cardinal and Pronghorn have them. the main motivation of creating this website is to avoid confusion in our users. Would you be opened to include Cardinal as the LES and DNS capability?

**Comment by joshuahansel** [comment]
> I am a bit worried about duplication of capabilities for RANS

If Cardinal provides RANS, think I think it should be listed that way. We just need to provide users with some guidance on which to pick for their situation.

**Comment by aprilnovak** [comment]
I'd lean towards keeping RANS in the list for NekRS, since it is a major aspect of NekRS's use cases. Perhaps these tables can add another column for "numerical method", and then part of the distinction between NekRS vs. Pronghorn would be spectral element vs. finite volume?

**Comment by tanoret** [comment]
Ok, as you all wish. Here is the question that bugs me. Which RANS capability should an external engineer (usually unfamiliar with the details of FE, FV, or spectral methods) choose when both Moose NS and Cardinal report RANS capabilities? This duplication of capabilities has been reducing adoption. So, as long as this question is resolved and clear in the website, I am good with any modifications

**Comment by licharlot** [comment]
We could use the triangle with the fidelity levels and add the modules on that scale. It would show that Cardinal is RANS/LES/DNS while NS is coarse mesh/RANS.

**Comment by aprilnovak** [comment]
I like the scale triangle/hierarchy portrayal idea. 👍 

I think there's some benefit to be had if a user is only interested in RANS/LES/DNS, that they can quickly see that they can learn a single tool (Cardinal). Conversely, if a user is only interested in RANS/porous media/subchannel, then they can quickly see that they can learn a single (different) tool (Pronghorn). If they want a full range of TH capabiliteis spanning subchannel up through DNS, then it'll be up to them to decide in which application they do the RANS modeling.

FYI SAM also has some capabilities which overlap those in Pronghorn (3-D porous media, for instance), which I don't think are listed in the table. If we're going to be comprehensive with the multiscale pyramid, these would be good to include.

**Comment by tanoret** [comment]
I think SAM 3D porous-media capabilities are now part of the MOOSE NS module. So, they should be included in the table.

To be more precise, let's say a customer wants to perform studies on an MSR using RANS modeling for reactor safety studies. I think that in this case they should use MOOSE NS since it couples more naturally to Girffin, BISON, etc. and has a large(r) validation base behind it. However, if they want to study the flow phenomena in the MSRs, while being able to deal with multiple levels of resolution in the flow field, then Cardinal is a better idea. I just don't see this clearly in the current table

**Comment by aprilnovak** [comment]
Well, Cardinal allows NekRS to be coupled to any other Moose App with just as much ease as Moose NS couples to other Moose Apps (not that this makes the explanation more straightforward, but wanted to note it). I don't have an easy answer - there's some overlap in the codes.

**Comment by tanoret** [comment]
Ok. Let's keep it as it is now. 

Unless someone has any idea on how to make things more clear, I guess we can always explain more about the capabilities of each of the codes if asked and the links to the documentation of each of the codes will provide more information as needed. Also, at some point, we should link the V&V matrices for each of the codes so that users can see past validation cases for more information.

**Comment by lindsayad** [comment]
I can't see the figures on the rendered documentation site? Does anyone else have this issue? My two theories are dark mode or zscaler

**Comment by joshuahansel** [comment]
> I can't see the figures on the rendered documentation site. Does anyone else have this issue? My two theories are dark mode or zscaler

Same for me. It's just a build failure because the pics aren't available yet. @tanoret must just have them locally or did not update large_media yet.

**Comment by tanoret** [comment]
I did and I think that they are online (https://github.com/idaholab/large_media/tree/master/thermal_hydraulics/misc). Anything I missed?

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+| [Scalar Kernels System](syntax/ScalarKernels/index.md)                  | 0D ODE                                 | Incompressible                                                                                                                                               | 0D                                                                 | 10                    | 1 second        | Bernoulli-like formulations for pressure drop                                                                                                                                                          |
+
+## Open-Source Applications for Advanced Engineering Modeling
+
```
Comment (grmnptr) [inline modules/doc/content/applications/thermal_hydraulics.md:24]:
```suggestion
## Open-Source MOOSE-based Wrappers for External Fluid Dynamics Applications

```

Comment (grmnptr) [reply modules/doc/content/applications/thermal_hydraulics.md:24]: How about this change? I agree that Cardinal is a very useful interface between Nek and everything MOOSE-based. But Nek is not MOOSE-based, and if I understand @tanoret 's proposal we are trying to advertise the tools we are working on a website we have more control over (compared to the NEAMS website). Adding advanced engineering modeling might make some visitors think that the tools mentioned before are not advanced and not for engineering.

**Comment by lindsayad** [comment]
I wonder if this page should only be about capabilities that a user gets when they clone MOOSE. As @tanoret has stated or implied, if we supply a lot of information on a single page it can overwhelm someone. It's like when you go to a restaurant and there are many items on the menu, you're overwhelmed and don't know what to pick. Perhaps it would be better to link to another page to see MOOSE-based applications that require additional effort from the user beyond cloning MOOSE

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+| Application Name                                               | Based On                  | Added Features                                                                                                                                         |
+| :------------------------------------------------------------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Pronghorn](https://pronghorn-dev.hpc.inl.gov/site/index.html) | Navier-Stokes Module      | Export controlled correlations for pressure drop, heat exchange, and mass transfer in advanced nuclear reactors.                                       |
+| SAM                                                            | MOOSE Framework           | Additional lumped parameters components for realistic plant modeling and proprietary correlations for pressure drop, heat exchange, and mass transfer. |
```
Comment (travismui) [inline modules/doc/content/applications/thermal_hydraulics.md:39]: @joshuahansel Thanks for putting this page together! Can you include this link for SAM for the time being? We'll be working on updating the webpage soon, but at least it has @rui-hu's contact info. @lingzou FYI

```suggestion
| [SAM](https://www.anl.gov/nse/system-analysis-module)                                                            | MOOSE Framework           | Additional physics and component models for realistic plant modeling and reactor safety analysis.  |
```

Comment (lindsayad) [reply modules/doc/content/applications/thermal_hydraulics.md:39]: thanks for the link @travismui !

**Comment by tanoret** [comment]
Are we good to go with this site? Any unresolved issues?

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+### High Temperature Reactors
+
+!media thermal_hydraulics/misc/example_httf.png
+       style=width:90%;display:block;margin-left:auto;margin-right:auto;
+       caption=Steady-state operation of Oregon State University's High Temperature Test Facility (HTTF) using the MOOSE Navier-Stokes module coarse-mesh CFD capability.
+       id=httf
+
+!media thermal_hydraulics/misc/example_httr.png
+       style=width:90%;display:block;margin-left:auto;margin-right:auto;
+       caption=Temperature in fuel blocks of the High-Temperature Engineering Test Reactor (HTTR) during steady-state operation using the MOOSE Navier-Stokes module coarse-mesh CFD capability.
```
Comment (licharlot) [inline modules/doc/content/applications/thermal_hydraulics.md:72]:
```suggestion
       caption=Temperature in fuel blocks of the High-Temperature Engineering Test Reactor (HTTR) during steady-state operation using the MOOSE Thermal Hydraulics Module (THM).
```

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+| :---------------------------------------------------------------------- | :------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------------------------------- | :-------------------- | :-------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Navier-Stokes Modue](navier_stokes/index.md)                           | Coarse-Mesh CFD \\ \\ RANS simulations | Incompressible, Weakly-Compressible, or Fully-Compressible\\ \\ Single- or Multi-Phase \\ \\ Single- or Multi-Component Flow                                 | Typically, 2D, 2D axisymmetric, or 3D \\ \\ Can also be used in 1D | 10,000                | 1 minute        | Flow through nuclear reactor core or plena \\ \\ 3D multi-phase flow in pipes \\ \\ Natural convection flow in open cavities                                                                           |
```
Comment (licharlot) [inline modules/doc/content/applications/thermal_hydraulics.md:17]:
```suggestion
| [Navier-Stokes Module](navier_stokes/index.md)                           | Coarse-Mesh CFD \\ \\ RANS simulations | Incompressible, Weakly-Compressible, or Fully-Compressible\\ \\ Single- or Multi-Phase \\ \\ Single- or Multi-Component Flow                                 | Typically, 2D, 2D axisymmetric, or 3D \\ \\ Can also be used in 1D | 10,000                | 1 minute        | Flow through nuclear reactor core or plena \\ \\ 3D multi-phase flow in pipes \\ \\ Natural convection flow in open cavities                                                                           |
```

**Review by licharlot — COMMENTED** [review]
Just a few typos.

File: modules/doc/content/applications/thermal_hydraulics.md (hunk 5 above)
Comment (GiudGiud) [reply modules/doc/content/applications/thermal_hydraulics.md:17]: the run time and elements could be split ?
I dont like saying we do 1 minute runs with 10k elements
maybe we can have
10 k.                   1 minute
>100k                Setup-dependent

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+| [Subchannel Module](https://subchannel-dev.hpc.inl.gov/site/index.html) | Subchannel Scale                       | Incompressible or Weakly-Compressible \\ \\ Single-Phase \\ \\ Single- or Multi-Component Flow                                                               | Typically, 3D \\ \\ Can be used in 1D and 2D                       | 100,000               | 10 seconds      | Flow development through nuclear reactor fuel assembly \\ \\ Thermal-hydraulics analysis of nuclear reactor assembly blockage \\ \\ Natural convection cooling in nuclear reactors low-flow assemblies |
```
Comment (GiudGiud) [inline modules/doc/content/applications/thermal_hydraulics.md:18]: this does not exist yet

add : To be released 2024

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+| :------------------------------------------------------------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Pronghorn](https://pronghorn-dev.hpc.inl.gov/site/index.html) | Navier-Stokes Module      | Export controlled correlations for pressure drop, heat exchange, and mass transfer in advanced nuclear reactors.                                       |
+| SAM                                                            | MOOSE Framework           | Additional lumped parameters components for realistic plant modeling and proprietary correlations for pressure drop, heat exchange, and mass transfer. |
+| [Sockeye](https://sockeye-dev.hpc.inl.gov/site/)               | Thermal-Hydraulics Module | Adapted correlations and specific 1D and 2D models for high-temperature heat pipes.                                                                    |
```
Comment (GiudGiud) [inline modules/doc/content/applications/thermal_hydraulics.md:40]: add RELAP7

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
+# Thermal Hydraulics Applications
+
+MOOSE provides versatile, general-purpose thermal-hydraulics applications. These applications solve for mass, momentum, and energy conservation in multicomponent, multiphase flows using incompressible, weakly-compressible, or fully compressible formulations for steady-state or transients in lumped parameters and/or multidimensional (1, 2, or full 3D) geometries.
```
Comment (GiudGiud) [inline modules/doc/content/applications/thermal_hydraulics.md:3]:
```suggestion
MOOSE provides versatile, general-purpose thermal-hydraulics applications. These applications solve for mass, momentum, energy, and advected species conservation in multicomponent, multiphase flows using incompressible, weakly-compressible, or fully compressible formulations for steady-state or transients in lumped parameters and/or multidimensional (1, 2, or full 3D) geometries.
```

**Review by GiudGiud — COMMENTED** [review]
good stuff

**Comment by lindsayad** [comment]
My vote was to split the applications off of this page, but I'm not going to hold up this PR if you guys are satisfied with it as-is

**Comment by GiudGiud** [comment]
You can ditch the large_media submodule update commit now. It was added by another PR to moose and your pictures are available

File: modules/doc/content/applications/thermal_hydraulics.md (hunk 8 above)
Comment (kyriv1980) [reply modules/doc/content/applications/thermal_hydraulics.md:3]: MOOSE itself is general purpose, the provided thermal-hydraulics applications are each quite specific. I propose the following:

MOOSE provides multiple thermal-hydraulic applications for varied purposes. Cumulatively, these applications solve for mass, momentum, energy, and advected species conservation in multicomponent, multiphase flows using incompressible, weakly-compressible, or fully compressible formulations for steady-state or transients in lumped parameters and/or multidimensional (1, 2, or full 3D) geometries.

File: modules/doc/content/applications/thermal_hydraulics.md (hunk 6 above)
Comment (kyriv1980) [inline modules/doc/content/applications/thermal_hydraulics.md:18]: Can be used in 1D and 2D makes no sense for subchannel. I would suggest calling it 2.5D or pseudo-3D or 3D.

Comment (tanoret) [reply modules/doc/content/applications/thermal_hydraulics.md:18]: As we discussed in the INL TH meeting in 03/06/24, we set for including only 3D

**Comment by joshuahansel** [comment]
Rebasing to get rid of the large media issue FYI.

File: modules/doc/content/applications/thermal_hydraulics.md (hunk 3 above)
Comment (joshuahansel) [reply modules/doc/content/applications/thermal_hydraulics.md]: Thanks! That's good that SAM has a public-facing page - we need that for some of the other apps.

File: modules/doc/content/applications/thermal_hydraulics.md (hunk 2 above)
Comment (joshuahansel) [reply modules/doc/content/applications/thermal_hydraulics.md]: I combined all into one table now, so this suggestion probably doesn't apply anymore.

**Comment by joshuahansel** [comment]
Ok I believe I've at least attempted to address everyone's concerns, so this is ready for review now.

**Comment by joshuahansel** [comment]
All, I've clicked resolve on the comments that I'm confident I addressed. Please check if each of your comments have been resolved (or if you're ok as-is) and then click resolve.

File: modules/doc/content/applications/thermal_hydraulics.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+are developed as part of the
+[Nuclear Energy Advanced Modeling and Simulation (NEAMS) program](https://neams.inl.gov/).
+Some of these applications are open-source, whereas some are export-controlled
+and distributed via the [Nuclear Computational Resource Center (NCRC)](https://inl.gov/ncrc/);
+see [help/inl/applications.md] for more information.
+
+| Application Name                                               | Distribution | Based On                  | Added Features                                                                                                                                         |
+| :------------------------------------------------------------- | :----------- | :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [Cardinal](https://cardinal.cels.anl.gov/) | Open-source | [NekRS](https://github.com/Nek5000/nekRS) CFD | CPU and GPU capabilities for RANS, LES, and DNS. Additional features include Lagrangian particle transport, an ALE mesh solver, overset meshes, and more. |
```
Comment (lindsayad) [inline modules/doc/content/applications/thermal_hydraulics.md:32]:
```suggestion
| [Cardinal](https://cardinal.cels.anl.gov/) | [Open-source](https://github.com/neams-th-coe/cardinal) | [NekRS](https://github.com/Nek5000/nekRS) CFD | CPU and GPU capabilities for RANS, LES, and DNS. Additional features include Lagrangian particle transport, an ALE mesh solver, overset meshes, and more. |
```

**Review by lindsayad — APPROVED** [review]
You guys decided to just do one page? Well I think this is definitely an improvement. Good work all

File: modules/doc/content/applications/thermal_hydraulics.md (hunk 6 above)
Comment (joshuahansel) [reply modules/doc/content/applications/thermal_hydraulics.md]: I couldn't remember what we decided on this, so I didn't do anything yet. Is subchannel in the process already? I just don't want to forget to remove that comment when it does get added.

Comment (joshuahansel) [reply modules/doc/content/applications/thermal_hydraulics.md]: I had written "1D with lateral coupling" - please change to "3D" if you prefer that

Comment (joshuahansel) [reply modules/doc/content/applications/thermal_hydraulics.md]: Changed to 3D

**Comment by GiudGiud** [comment]
Front page broken for me since this got merged
anyone else?
@milljm I remember you tried a couple platforms for another PR

Screenshot 2024-03-12 at 10 12 48 AM

**Comment by lindsayad** [comment]
what browser? It's been working well for me on chrome with dark mode

**Comment by GiudGiud** [comment]
Chrome with dark mode too

**Comment by joshuahansel** [comment]
So basically it's a problem with your window size being too small, since we added some length to it. Either need smarter resizing or we need to lose some weight on that header bar.

**Comment by GiudGiud** [comment]
yes most likely

**Pull request merged**